File: frontend/src/types/product.ts
export interface Product {
    id: number;
    title: string;
    description: string;
    price: number;
    category: string;
    sellerName: string;
    imageUrl: string | null;
    postedDate: string;
}

export interface ProductResponse {
    id: number;
    title: string;
    description: string;
    price: number;
    category: string;
    sellerName: string;
    imageUrl: string | null;
    postedDate: string;
}

export interface ProductFilters {
    category?: string;
    minPrice?: number;
    maxPrice?: number;
}

File: backend/Models/Product.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Models;

public class Product
{
    public int Id { get; set; }

    [Required, MaxLength(200)]
    public string Title { get; set; } = string.Empty;

    [MaxLength(2000)]
    public string Description { get; set; } = string.Empty;

    [Column(TypeName = "decimal(12,2)")]
    public decimal Price { get; set; }

    public int CategoryId { get; set; }
    public Category Category { get; set; } = null!;

    [Required, MaxLength(100)]
    public string SellerName { get; set; } = string.Empty;

    public string? ImageUrl { get; set; }

    public DateTime PostedDate { get; set; } = DateTime.UtcNow;
}
